src/main/java/org/jfree/svg/util/RyuDouble.java

```diff
@@ -0,0 +1,415 @@
+          break;
+        }
+        lastRemovedDigit = (int) (dv % 10);
+        dp /= 10;
+        dv /= 10;
+        dm /= 10;
+        removed++;
+      }
+      output = dv + ((dv == dm || (lastRemovedDigit >= 5)) ? 1 : 0);
+    }
+    int olength = vplength - removed;
+
+    // Step 5: Print the decimal representation.
+    // We follow Double.toString semantics here.
+    char[] result = new char[24];
+    int index = 0;
+    if (sign) {
+      result[index++] = '-';
+    }
+
+    // Values in the interval [1E-3, 1E7) are special.
+    if (scientificNotation) {
+      // Print in the format x.xxxxxE-yy.
+      for (int i = 0; i < olength - 1; i++) {
+        int c = (int) (output % 10); output /= 10;
+        result[index + olength - i] = (char) ('0' + c);
+      }
+      result[index] = (char) ('0' + output % 10);
+      result[index + 1] = '.';
+      index += olength + 1;
+      if (olength == 1) {
+        result[index++] = '0';
+      }
+
+      // Print 'E', the exponent sign, and the exponent, which has at most three digits.
+      result[index++] = 'E';
+      if (exp < 0) {
+        result[index++] = '-';
+        exp = -exp;
+      }
+      if (exp >= 100) {
+        result[index++] = (char) ('0' + exp / 100);
+        exp %= 100;
+        result[index++] = (char) ('0' + exp / 10);
+      } else if (exp >= 10) {
+        result[index++] = (char) ('0' + exp / 10);
+      }
+      result[index++] = (char) ('0' + exp % 10);
+      return new String(result, 0, index);
+    } else {
+      // Otherwise follow the Java spec for values in the interval [1E-3, 1E7).
+      if (exp < 0) {
+        // Decimal dot is before any of the digits.
+        result[index++] = '0';
+        result[index++] = '.';
+        for (int i = -1; i > exp; i--) {
+          result[index++] = '0';
+        }
+        int current = index;
+        for (int i = 0; i < olength; i++) {
+          result[current + olength - i - 1] = (char) ('0' + output % 10);
+          output /= 10;
+          index++;
+        }
+      } else if (exp + 1 >= olength) {
+        // Decimal dot is after any of the digits.
+        for (int i = 0; i < olength; i++) {
+          result[index + olength - i - 1] = (char) ('0' + output % 10);
+          output /= 10;
+        }
+        index += olength;
+        for (int i = olength; i < exp + 1; i++) {
+          result[index++] = '0';
+        }
+        result[index++] = '.';
+        result[index++] = '0';
+      } else {
+        // Decimal dot is somewhere between the digits.
+        int current = index + 1;
+        for (int i = 0; i < olength; i++) {
+          if (olength - i - 1 == exp) {
+            result[current + olength - i - 1] = '.';
+            current--;
+          }
+          result[current + olength - i - 1] = (char) ('0' + output % 10);
+          output /= 10;
+        }
+        index += olength + 1;
+      }
+      return new String(result, 0, min( index, decimals + (sign ? 3 : 2) ));
```

Comment:
This was the most direct way to truncate the output, but possibly not the most efficient. The code appears to loop a few places to generate the digit string after the decimal. It may be more efficient to terminate those loops early instead of truncating at the end.